lib/elixir/lib/path.ex

```diff
@@ -687,7 +687,19 @@ defmodule Path do
     case {rest, major_os_type()} do
       {"\\" <> _, :win32} -> System.user_home!() <> rest
       {"/" <> _, _} -> System.user_home!() <> rest
-      _ -> "~" <> rest
+      _ -> expand_user_home(rest)
+    end
+  end
+
+  defp expand_user_home(from_user) do
+    [user | rest] = String.split(from_user, "/")
+    rest = Enum.join(rest, "/")
+    user_home = "/home/" <> user
```

Comment:
We probably need to use `System.user_home!` instead of "/home/".

Reply:
I was trying to get the home for any user, not only for the current one.